rocAL/source/loaders/audio/audio_read_and_decode.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright (c) 2019 - 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+
+#include <iterator>
+#include <cstring>
+#include "decoder_factory.h"
+#include "audio_decoder_factory.h"
+#include "audio_read_and_decode.h"
+
+Timing
+AudioReadAndDecode::timing() {
+    Timing t;
+    t.audio_decode_time = _decode_time.get_timing();
+    t.audio_read_time = _file_load_time.get_timing();
+    return t;
+}
+
+AudioReadAndDecode::AudioReadAndDecode():
+    _file_load_time("FileLoadTime", DBG_TIMING ),
+    _decode_time("DecodeTime", DBG_TIMING) {
+}
+
+AudioReadAndDecode::~AudioReadAndDecode() {
+    _reader = nullptr;
+    _decoder.clear();
+}
+
+void
+AudioReadAndDecode::create(ReaderConfig reader_config, DecoderConfig decoder_config, int batch_size, int device_id) {
+    // Can initialize it to any decoder types if needed
+    _batch_size = batch_size;
+    _decoder.resize(batch_size);
+    _audio_names.resize(batch_size);
+    _audio_file_path.resize(batch_size);
+    _decompressed_buff_ptrs.resize(_batch_size);
+    _actual_decoded_samples.resize(_batch_size);
+    _actual_decoded_channels.resize(_batch_size);
+    _original_channels.resize(_batch_size);
+    _original_samples.resize(_batch_size);
+    _original_sample_rates.resize(_batch_size);
+    _decoder_config = decoder_config;
+    if ((_decoder_config._type != DecoderType::SKIP_DECODE)) {
+        for (int i = 0; i < batch_size; i++) {
+            _decoder[i] = create_audio_decoder(decoder_config);
+        }
+    }
+    _reader = create_reader(reader_config);
+    _input_path = reader_config.path();
```

Comment:
Is this particular variable used anywhere I could see we are just getting this and not using please check

Reply:
removed as its not used